### ISSUE TITLE
Support @apollo-react/hooks

### ISSUE THREAD

**Issue by slikts** [dotansimha/graphql-code-generator#1863]
The official hooks support for React Apollo [has been released](https://twitter.com/peggyrayzis/status/1128280687470100480?s=20) as a beta version, and it'd be nice to have support for it in gql-gen. The API is close to react-apollo-hooks, which is already supported, so it shouldn't take major changes (probably fewer changes than in #1814, although I've only looked at it in passing).

**Comment by ardatan** [comment]
It's supposed to work with `hooksImportFrom: @apollo-react/hooks` configuration. We can keep this issue until it is confirmed.

**Comment by SimenB** [comment]
The types are wrong (typed as `QueryResult<any, any>`: https://unpkg.com/@apollo/react-hooks@0.1.0-beta.2/lib/useQuery.d.ts). Seems like the shared code sandbox (https://codesandbox.io/s/github/peggyrayzis/redux-to-graphql/tree/master/client) uses `const res: QueryResult<ApolloTypes.GetSubreddit, ApolloTypes.GetSubredditVariables> = useQuery()` instead. Not sure if that's on purpose or not, but it means that the generated code is not correct

**Comment by ardatan** [comment]
I think it is better to wait the stable release. @dotansimha What do you think?

**Comment by SimenB** [comment]
Might be nice to release an alpha or something supporting it (it's just moving the generics a bit and adding an import), just so people can test it? Completely understand if you don't wanna put work into it though, it seems silly (probably an oversight) to not allow passing in the generics at the callsite

**Comment by dotansimha** [comment]
I think it worth to add the option to use it, maybe by just doing: `hooksImportFrom: @apollo-react/hooks` we can change the output to match their signature.

**Comment by SimenB** [comment]
I asked: https://github.com/apollographql/react-apollo/pull/2892#issuecomment-492949070

**Comment by ardatan** [comment]
@SimenB As I see in the current state of that PR in apollo,
```tsx
export function useVoteMutation(baseOptions?: ReactApolloHooks.MutationHookOptions<VoteMutation, VoteMutationVariables>) {
  return ReactApolloHooks.useMutation<VoteMutation, VoteMutationVariables>(VoteDocument, baseOptions);
};
```
This generated code fits their signature since we don't specify any return type etc. We only pass generics and `Options` interface as a parameter that also fits Apollo's `Options` interface.

**Comment by SimenB** [comment]
The types are closer now. However, an actual difference is in `QueryHookOptions`.

in `@apollo/react-hooks` it is

```ts
export interface QueryHookOptions<TData = any, TVariables = OperationVariables> extends QueryFunctionOptions<TData, TVariables> {
    query?: DocumentNode;
}
```

while in `react-apollo-hooks` it is
```ts
export interface QueryHookOptions<TVariables, TCache = object> extends Omit<QueryOptions<TVariables>, 'query'> {
    notifyOnNetworkStatusChange?: boolean;
    pollInterval?: number;
    client?: ApolloClient<TCache>;
    ssr?: boolean;
    skip?: boolean;
    suspend?: boolean;
}
```

This means that the `baseOptions?: ReactApolloHooks.QueryHookOptions<QueryVariables>` that this project generates is wrong. Not sure in which of the 3 projects the fix should be in, though

**Comment by dotansimha** [comment]
@SimenB we can output a different signature when `importHooksFrom` is set to `@apollo/react-hooks` and keep the original output as-is. Then change the output if `@apollo/react-hooks` will introduce a new signature later.

**Comment by ardatan** [comment]
@dotansimha I think since react-apollo is for Apollo, we should make @apollo/react-hooks by default after the stable release, and create another plugin for `react-apollo-hooks`

**Comment by edorivai** [comment]
I've taken a stab at the implementation (#1962) @dotansimha mentioned [here](https://github.com/dotansimha/graphql-code-generator/issues/1863#issuecomment-496586940).

So this will change the output if `importHooksFrom` is set to `@apollo/react-hooks`.

**Comment by NickClark** [comment]
The latest release works great guys! Except one little thing... I currently have to remove the mutation functions manually since my project depends on the new "@apollo/react-common" and has HOC and components turned off. Turns out if you only have hooks turned on anyways, that I don't need the "@apollo/react-common" import or the Mutation functions anyways. 

Said another way. If you have your project with only hooks enabled, and using "@apollo/react-hooks", then we shouldn't import "react-apollo", "@apollo/react-common", or generate mutations functions

**Comment by dotansimha** [comment]
You are right, @NickClark , the `MutationFn` are not really required, and so does the imports, but you can turn it off using `withMutationFn: false`).

**Comment by dotansimha** [comment]
I think we can prepare to close this one. We currently support `@apollo-react/hooks` (2/3) and `apollo-react-hooks`.

**Comment by dotansimha** [comment]
Available in 1.4.0 🎉
